ccl/changefeedccl: TestChangefeedOrderingWithErrors failed

ccl/changefeedccl.TestChangefeedOrderingWithErrors [failed](https://tanzanite.cluster.engflow.com/invocations/default/03743512-f206-4e98-bdb8-59ecfd62b924?testReportRun=3&testReportShard=7&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.1 @ [6e3e9e6012dc04dabc64333a87e77a80d9f9d46b](https://github.com/cockroachdb/cockroach/commits/6e3e9e6012dc04dabc64333a87e77a80d9f9d46b):


```
=== RUN   TestChangefeedOrderingWithErrors
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedOrderingWithErrors2292631964
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_test.go:7126: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedOrderingWithErrors2292631964
--- FAIL: TestChangefeedOrderingWithErrors (315.23s)
=== RUN   TestChangefeedOrderingWithErrors/webhook
    helpers_test.go:867: making server as system tenant
    helpers_test.go:943: making webhook feed factory
    changefeed_test.go:7117: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:199
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:286
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_test.go:7117
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1104
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:1139
        	Error:      	Received unexpected error:
        	            	operation "withTimeout-960444964341710849" timed out after 5m0.001s (given timeout 5m0s): stopping job due to TestFeed timeout
        	            	(1) operation "withTimeout-960444964341710849" timed out after 5m0.001s (given timeout 5m0s)
        	            	Wraps: (2) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func3
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:2838
        	            	  | github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl.stopFeedWhenDone.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/testfeed_test.go:2827
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1695
        	            	Wraps: (3) stopping job due to TestFeed timeout
        	            	Error types: (1) *timeutil.TimeoutError (2) *withstack.withStack (3) *errutil.leafError
        	Test:       	TestChangefeedOrderingWithErrors/webhook
    --- FAIL: TestChangefeedOrderingWithErrors/webhook (315.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=7</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedOrderingWithErrors.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-37831

--- a/pkg/ccl/changefeedccl/cdctest/mock_webhook_sink.go
+++ b/pkg/ccl/changefeedccl/cdctest/mock_webhook_sink.go
@@ -114,6 +114,7 @@ func (s *MockWebhookSink) SetStatusCodes(statusCodes []int) {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	s.mu.statusCodes = statusCodes
+	s.mu.statusCodesIndex = 0
 }
 
 // Close closes the mock Webhook sink.

